### DISCUSSION TITLE
Modelina 1. year roadmap

### DISCUSSION THREAD

**Discussion by jonaslagoni** [asyncapi/modelina#848] [Ideas]
Now that version 1, is creeping closer and closer, I thought it was time to start a discussion around what we should focus on going forward. I have tried to align this roadmap against the [AsyncAPI roadmap](https://www.asyncapi.com/roadmap) to have some sense of root in AsyncAPI.

Of course, this roadmap does NOT limit your contributions or even what you would like to work on. This will be a north star that will be in focus and prioritized. So the roadmap forms an alignment of thoughts toward that single vision, without limiting natural change over time.

# Vision 
Modelina becomes the GOTO solution for generating models for APIs.

## Goals
What are we fighting for?

- 250 contributors and at least [two champions, per area of responsibility](https://github.com/asyncapi/modelina/blob/master/docs/champions.md) :handshake:
- At least three ways to serialize models :three:
- Regardless of your input, Modelina supports it :100:

### 250 contributors and at least two champions per area of responsibility :handshake:
Because every contributor has a limited time they can spend on Modelina, I feel it's vital that we strive towards sharing the responsibility as much as possible. This is not a one-person army, it's a shared, collaborative project.

That's why I suggest one of the goals focuses on the champions and contributors.

But this goal might seem simple on the surface, but it's nowhere near as easy as it sounds. We have to enable as broad a spectrum of contributions as possible, this includes but not limited to blogs, tweets, videos, examples, documentation, playground, and mentorships, easy for first-time contributors, and challenging for experienced ones.

Therefore I am proposing that all of the roadmap goals and tasks will be created in the form of good first issues that will enable you to contribute to Modelina, regardless of whether you like to code, write kickass documentation or blog posts, or create stunning videos... You name it you can contribute!

### At least three ways to serialize models :three:
JSON is not everything in this world there can be times when you want your payload models to be serialized to XML or even binary. You might even want to use different libraries as each has strengths and weaknesses.

The user should be able to decide for themself what they would like to use. 

Now keep in mind this has to be replicated for each code generator, which requires knowledge of the output language, hence goal 1.

### Regardless of your input, Modelina supports it :100:
Here I want to focus on the inputs you can provide Modelina. Here I am thinking about JSON Schema 2019-09, JSON Schema 2020-12, Avro, JSON Type definition, GraphQL Schema, Protobuf, OpenAPI 3.1, Postman collections, etc. 

What triggers this goal, is that AsyncAPI supports an unlimited amount of schema formats, and to support them in code templates (that integrate Modelina), we must extend the current inputs to support many different formats.

## Feedback
This roadmap will directly influence the future of Modelina, so any feedback is welcomed! 

Do not see this vision as set in stone, would love counter ideas and discussions around what would be best suited for the next year for Modelina, so don't hold back :muscle:

Once the discussion is settled and the roadmap is aligned I will make sure to finalize it into an issue that can be used as a north star going forward.

cc. current champions @magicmatatjahu @ron-debajyoti @Samridhi-98 @leigh-johnson

**Comment by handrews** [comment]
I would (as a personal opinion, [not currently a JSON Schema org position](https://github.com/orgs/json-schema-org/discussions/209)) recommend against too much effort on JSON Schema draft 2019-09.  It was not out long enough to get broadly adopted before 2020-12, which fixes some problems and has OpenAPI 3.1 driving further adoption.

**Reply by jonaslagoni** [reply]
It's just input, so there will not be any special focus on one version or the other tbh 🙂 The "problem" is that once the draft is released, it needs to have the possibility to be supported 🙂 From an AsyncAPI perspective it is definitely a possibility that users can use those versions of schemas (through `schemaFormat`).

But yea, definitely going to keep it in mind when spending time on it 👍  

(Just got back from holiday, sorry for the delayed response)
